codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -756,7 +756,14 @@ public ChannelFuture resetStream(final ChannelHandlerContext ctx, int streamId,
             return promise.setSuccess();
         }
 
-        ChannelFuture future = frameWriter().writeRstStream(ctx, streamId, errorCode, promise);
+        final ChannelFuture future;
+        if (connection().local().created(stream) && !stream.isFrameSent()) {
```

Comment:
When we send `HEADERS`, we're still creating it in the open state so there's no easy way to check if the `HEADERS` have actually been sent.